### PR TITLE
Add `document` activity dataviz

### PR DESCRIPTION
**Pull request by quitterie-lcs**
## Purpose

Document is a ressource commonly used in e-learning. It is necessary to get to
know how many times a document is downloaded.

## Proposal

- [x] Add indicators backend plugin
- [x] Add dashboard frontend package

### CODE REVIEW

File: src/frontend/packages/document/src/components/Datagrid/index.tsx
Comment (wilbrdt) [inline src/frontend/packages/document/src/components/Datagrid/index.tsx:1]: As the `Document` components are almost identical to the `Video` components, I was wondering if we shouldn't make them generic into the `core` frontend package

File: src/frontend/apps/web/src/App.tsx
Comment (wilbrdt) [inline src/frontend/apps/web/src/App.tsx:1]: Out of scope of this PR, but while a Document-only dashboard is a good starting point, we might want to consider having its component in a more general dashboard with visualization about other activities.

File: src/api/plugins/document/warren_document/factories.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "display": {"en-US": "downloaded"},
+        },
+        "context": {
+            "extensions": {
+                "https://w3id.org/xapi/video/extensions/session-id": (
+                    "6e952c73-71e3-5c8c-837b-39baea0e73b9"
+                ),
+                "https://w3id.org/xapi/video/extensions/quality": 720,
+                "https://w3id.org/xapi/video/extensions/length": 508,
+            },
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/api/plugins/document/warren_document/factories.py:22]: Good catch, thanks !

File: src/frontend/packages/document/src/components/DocumentFilters/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  { id: "uuid://1c0c127a-f121-4bd1-8db6-918605c2645d", title: "Lesson 2" },
+  { id: "uuid://541dab6b-50ae-4444-b230-494f0621f132", title: "Lesson 3" },
+  { id: "uuid://69d32ad5-3af5-4160-a995-87e09da6865c", title: "Lesson 4" },
+  { id: "uuid://7d4f3c70-1e79-4243-9b7d-166076ce8bfb", title: "Lesson 5" },
+  { id: "uuid://8d386f48-3baa-4acf-8a46-0f2be4ae243e", title: "Lesson 6" },
+  { id: "uuid://b172ec09-97ec-4651-bc57-6eabebf47ed0", title: "Lesson 7" },
+  { id: "uuid://d613b564-5d18-4238-a69c-0fc8cee5d0e7", title: "Lesson 8" },
+  { id: "uuid://dd38149d-956a-483d-8975-c1506de1e1a9", title: "Lesson 9" },
+  { id: "uuid://e151ee65-7a72-478c-ac57-8a02f19e748b", title: "Lesson 10" },
+];
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/frontend/packages/document/src/components/DocumentFilters/index.tsx:24]: I will address this point in another PR.

File: src/frontend/packages/document/src/components/Datagrid/index.tsx
Comment (quitterie-lcs) [reply src/frontend/packages/document/src/components/Datagrid/index.tsx:1]: I will address this point in a dedicated PR.

File: src/frontend/apps/web/src/App.tsx [hunk 3]
```diff
@@ -23,8 +23,22 @@ const getVideoRoute = (): React.LazyExoticComponent<() => JSX.Element> => {
...
+  student: lazy(() => import("@openfun/warren-document/src/pages/Student")),
+};
+
+const getDocumentRoute = (): React.LazyExoticComponent<() => JSX.Element> => {
+  if (dataContext.is_instructor) {
+    return documentRoutes.instructor;
+  } else {
+    return documentRoutes.student;
+  }
+};
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/frontend/apps/web/src/App.tsx:37]: I don't have an idea for now. I think we should brainstorm and propose something new in another PR.

[the remark this answers is not in the document]
Comment (wilbrdt) [reply src/frontend/apps/web/src/App.tsx:37]: I'm OK with that!

**Pull request merged**